src/DynamoCore/Engine/NodeToCode/NodeToCode.cs

```diff
@@ -320,6 +321,55 @@ public ShortestQualifiedNameReplacer(ClassTable classTable, ElementResolver reso
             this.resolver = resolver;
         }
 
+        public override AssociativeNode VisitTypedIdentifierNode(TypedIdentifierNode node)
+        {
+            if (node == null)
+                return null;
+
+            // If type is primitive type
+            if (node.datatype.UID != (int)PrimitiveType.InvalidType &&
+                node.datatype.UID < (int)PrimitiveType.MaxPrimitive)
+                return node;
+            
+            // build an idlistnode from the full type name
+            var identListNode = CoreUtils.CreateNodeFromString(node.datatype.Name);
+            if (identListNode == null)
+                return null;
+
+            // visit the idlist 
+            var shortNameNode = identListNode.Accept(this);
+            if(shortNameNode == null)
+                return null;
+            
+
+            var resultingNode = new TypedIdentifierNode();
+            resultingNode.Name = node.Name;
+            resultingNode.Value = node.Name;
+            resultingNode.datatype = node.datatype;
+
+            //return a typedIdNode built from the shortNameId returned above.
+            if (shortNameNode is IdentifierListNode)
+            {
+                var idListNode = (shortNameNode as IdentifierListNode);
+                resultingNode.TypeAlias = idListNode.ToString();
+            }
+
+            else if (shortNameNode is IdentifierNode)
+            {
+                var idNode = (shortNameNode as IdentifierNode);
+                resultingNode.TypeAlias = idNode.Value;
+            }
+
+            //modify the AST node that was passed to the visitor.
+            node.Name = resultingNode.Name;
+            node.Value = resultingNode.Value;
+            node.datatype = resultingNode.datatype;
+            node.TypeAlias = resultingNode.TypeAlias;
```

Comment:
Does `node` have to be passed as a `ref` parameter in order to directly assign it to `resultingNode`? Is that why you have modified its properties individually here?

Reply:
yep, I found that assigning `node = resultantNode` did not update the node I passed in, and using a ref parameter requires touching the ASTreplacer base class - so this is not ideal, but it does seem to work.